Rakefile

```diff
@@ -334,18 +293,15 @@ TESTS.each do |test|
         .find { |f| File.exist? f }
 
 
-      MUTEX.synchronize do
-        puts "#{Progress.inc_s} Compiling #{input} for #{engine}"
-      end
+      puts "#{Progress.inc_s} Compiling #{input} for #{engine}"
 
-      begin
-        File.write t.name, SM[endpoint].compile(input).clean
-      rescue SM::Error => e
-        MUTEX.synchronize do
-          STDERR.puts "  #{e} with #{input} for #{engine}"
-        end
+      result = `#{ENGINE_EXEC[engine]} #{input}`
+      unless $?.exitstatus > 0
+        File.write t.name, result.clean
+      else
+        STDERR.puts "  error occured with #{input} for #{engine}"
```

Comment:
You're correct. This is currently failing 4 tests include the error directive test. Using docker actually makes it's trivial to capture the error output so this can be easily dealt with.